docs/migration-guides/prometheus-migration.md

```diff
@@ -0,0 +1,328 @@
+
+```bash
+~ $ kubectl get pods -o=jsonpath='{range .items[*]}{"\n"}{.metadata.name}{":\t"}{range .spec.containers[*]}{.image}{", "}{end}{end}' | grep prometheus |\
+sort
+
+prometheus-0:	bitnami/prometheus:2.4.3-r31, jimmidyson/configmap-reload:v0.2.2, alpine:3.8,
+```
+
+Now, you can do a quick test of the rsync service by getting a shell in the container executing:
+
+`kubectl exec -it prometheus-0 -n kubeprod -c sidecar-rsync sh`
+
+and:
+
+`rsync -avrP prometheus-rsync.NAMESPACE.svc.cluster.local::data/test-file .`
+
+> Make sure previously you have deployed a test file in your Prometheus data directory so you can rsync it to the new container
+
+## Syncing with the compact times of the Prometheus database
+Now that all bits are in place you are ready to perform the Prometheus TSDB migration. To do so, you will first have to sync with the TSDB compact times of Prometheus to ensure the minimal data loss.
+
+### Doing a Prometheus TSDB migration
+
+By default, each two hours Prometheus will perform a compact of the TSDB and will write into disk the latest information it has in the WAL (Write-Ahead-Log) file. You can notice about these compacts in the TSDB by looking at the Prometheus logs:
+
+```bash
+level=info ts=2018-12-05T07:00:08.164203448Z caller=compact.go:398 component=tsdb msg="write block" mint=1543982400000 maxt=1543989600000 ulid=01CXYJN992AVWSVVJFP6X4E8MR
+level=info ts=2018-12-05T07:00:09.142253971Z caller=head.go:446 component=tsdb msg="head GC completed" duration=183.072583ms
+
+level=info ts=2018-12-05T09:00:08.289897499Z caller=compact.go:398 component=tsdb msg="write block" mint=1543989600000 maxt=1543996800000 ulid=01CXYSH0M8RSWNJG3YM1S9YGY5
+level=info ts=2018-12-05T09:00:09.242667991Z caller=head.go:446 component=tsdb msg="head GC completed" duration=170.063092ms
+```
+
+You can see how the delta between those two compacts is two hours.
+
+In order to do a safe migration avoiding block time ranges overlaps in next TSDB compacts after the migration, it will be necessary to create a snapshot in Prometheus skipping the WAL file, as the WAL file will contain references to future time ranges.
+
+It is important to know that if you create a Prometheus snapshot skipping the WAL file the *last hour of data* will not be present in the snapshot. In example:
+
+If I sync with the Prometheus TSDB and I see the latest logs from Prometheus are:
+
+```bash
+level=info ts=2018-12-05T11:00:08.289897499Z caller=compact.go:398 component=tsdb msg="write block" mint=1543989600000 maxt=1543996800000 ulid=01CXYSH0M8RSWNJG3YM1S9YGY5
+level=info ts=2018-12-05T13:00:09.242667991Z caller=head.go:446 component=tsdb msg="head GC completed" duration=170.063092ms
+```
+
+A good time to create a snapshot of the TSDB  to perform the migration would be at 13:00 UTC, so I create the snapshot just after Prometheus wrote the latest information stored in the WAL file to disk.
```

Comment:
Replace the first person (I) with the second person (you).

Reply:
done.